src/mlpack/tests/adaboost_test.cpp

```diff
@@ -716,16 +756,38 @@ BOOST_AUTO_TEST_CASE(ClassifyTest_IRIS)
     BOOST_FAIL("Cannot load test dataset iris_test.csv!");
 
   arma::Row<size_t> predictedLabels(testData.n_cols);
-
   a.Classify(testData, predictedLabels);
 
   arma::Mat<size_t> trueTestLabels;
   if (!data::Load("iris_test_labels.csv", trueTestLabels))
     BOOST_FAIL("Cannot load test dataset iris_test_labels.csv!");
 
+  arma::Row<size_t> predictedLabels1(testData.n_cols),
+                    predictedLabels2(testData.n_cols);
+  arma::mat probabilities;
+
+  a.Classify(testData, predictedLabels1);
+  a.Classify(testData, predictedLabels2, probabilities);
+
+  BOOST_REQUIRE_EQUAL(probabilities.n_cols, testData.n_cols);
+
+  for (size_t i = 0; i < predictedLabels1.n_cols; ++i)
+    BOOST_REQUIRE_EQUAL(predictedLabels1[i], predictedLabels2[i]);
+
+  arma::colvec pRow;
+  arma::uword maxIndex = 0;
+
+  for (size_t i = 0; i < predictedLabels1.n_cols; i++)
+  {
+    pRow = probabilities.unsafe_col(i);
+    pRow.max(maxIndex);
+    BOOST_REQUIRE_EQUAL(predictedLabels1(i), maxIndex);
+    BOOST_REQUIRE_EQUAL(arma::accu(probabilities.col(i)), 1);
```

Comment:
We can't compare floating-point types directly with `BOOST_REQUIRE_EQUAL()`; instead, use `BOOST_REQUIRE_CLOSE()` with a tolerance of, e.g., 1e-5, when comparing floating-point values. This should fix the test failures. :)